src/components/input-dropdown/input-dropdown.component.spec.ts

```diff
@@ -1,77 +1,200 @@
+    let component: InputDropdownTestComponent;
+    let fixture: ComponentFixture<InputDropdownTestComponent>;
+    let nativeElement: HTMLElement;
+    let overlayContainer: OverlayContainer;
+    let overlayContainerElement: HTMLElement;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [
+                InputDropdownModule,
+                NoopAnimationsModule,
+                RadioButtonModule
+            ],
+            declarations: [InputDropdownTestComponent]
+        }).compileComponents();
+
+        // access the overlay container
+        inject([OverlayContainer], (oc: OverlayContainer) => {
+            overlayContainer = oc;
+            overlayContainerElement = oc.getContainerElement();
+        })();
+
+        fixture = TestBed.createComponent(InputDropdownTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+    }));
+
+    afterEach(() => {
+        overlayContainer.ngOnDestroy();
+    });
+
+    it('should open dropdown when dropdownOpen is programmatically set to true', async () => {
+
+        component.dropdownOpen = true;
+
+        fixture.detectChanges();
+        await fixture.whenStable();
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [InputDropdownComponent],
-      imports: [CommonModule, FormsModule, IconModule, MenuModule]
-    })
-      .compileComponents();
-  }));
+        const dropdown = document.querySelector('.filter-container');
+        expect(dropdown).toBeTruthy();
+    });
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(InputDropdownComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    it('should close dropdown when dropdownOpen is programmatically set to false', async () => {
 
-    component.filterInputElement = new ElementRef({focus: () => true});
-  });
+        component.dropdownOpen = true;
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
-  });
+        fixture.detectChanges();
+        await fixture.whenStable();
 
-  it('should reset filter correctly', () => {
-    component.filterText = 'Tralala';
+        component.dropdownOpen = false;
 
-    spyOn(component.filterChange, 'emit');
-    spyOn(component.filterInputElement.nativeElement, 'focus');
+        fixture.detectChanges();
+        await fixture.whenStable();
 
-    component.resetFilter(new MouseEvent('click'));
+        const filter = document.querySelector('.filter-container');
+        expect(filter).toBeFalsy();
+    });
 
-    fixture.detectChanges();
-    expect(component.filterText).toEqual('');
-    expect(component.filterChange.emit).toHaveBeenCalledWith('');
-    expect(component.filterInputElement.nativeElement.focus).toHaveBeenCalledWith();
-  });
+    it('dropdownOpenChange should not emit when dropdownOpen is programmatically set', async () => {
```

Comment:
We should just test the emit in the following tests:
- "should open dropdown when dropdownOpen is programmatically set to true"
- "should close dropdown when dropdownOpen is programmatically set to false"

We shouldn't need a test that repeats previous tests with an extra check, we should just include the check in the previous tests